### PR TITLE
Add a checkbox to show all ports if desired 

### PR DESCRIPTION
**Pull request by Oats87**
Add a checkbox to show all ports if desired 
Screenshot 2024-09-10 at 5 19 48 PM

With the feature enabled:

Screenshot 2024-09-10 at 5 19 57 PM

With the feature disabled:

Screenshot 2024-09-10 at 5 20 02 PM

It will default to "off"

### CODE REVIEW

File: src/js/serial.js
Comment (breavyn) [inline src/js/serial.js:1]: This is leaking the GUI into the serial driver. Move the filter out into the port handler, either at the call site or perhaps in `updatePortSelect`.

**Review by breavyn — CHANGES_REQUESTED** [review]

**Comment by Oats87** [comment]
@breavyn I went ahead and moved the filter out (and am now just calling `.filter` on the array). 

I think we still need to figure out what to do about Android. We could hide the element and default it to "show all ports" if running on android (but I don't know how to reliably determine this) or something else...

File: src/js/port_handler.js [hunk 1]
```diff
@@ -255,11 +271,7 @@ PortHandler.selectPort = function(ports) {
         const portName = ports[i].displayName;
         if (portName) {
             const pathSelect = ports[i].path;
-            const isWindows = (OS === 'Windows');
```
Comment (breavyn) [inline src/js/port_handler.js:258]: `OS` is now unused, remove it.

File: src/js/port_handler.js
Comment (breavyn) [inline src/js/port_handler.js:1]: Fix the lint errors in this file. You can see them with  `yarn lint`.

**Review by breavyn — CHANGES_REQUESTED** [review]
When enabling show all, this is autoconnecting to a device if available. I don't think this is the correct behaviour.

For android, I think it will be best to ignore `show_all_ports` and hide the toggle widget for now. We can revisit this in the future when device names are working.

File: src/js/port_handler.js
Comment (Oats87) [reply src/js/port_handler.js:1]: File has been corrected to pass `yarn lint`

File: src/js/port_handler.js (hunk 1 above)
Comment (Oats87) [reply src/js/port_handler.js:258]: Removed `OS`

File: src/js/port_handler.js [hunk 2]
```diff
@@ -40,12 +42,35 @@ PortHandler.check = function () {
...
+};
+
 PortHandler.check_serial_devices = function () {
     const self = this;
 
     serial.getDevices(function(currentPorts) {
-
-        // auto-select port (only during initialization)
+        if (!self.showingAllPorts) {
+            currentPorts = currentPorts = currentPorts.filter((p) => portRecognized(p.displayName, p.path));
```
Comment (breavyn) [inline src/js/port_handler.js:71]: Reassign `currentPorts` once.

**Review by breavyn — CHANGES_REQUESTED** [review]
This is looking good. I've tested it behaves as expected on desktop and android.

Fix the `currentPorts` assignment and I'll approve. Also don't be afraid to split long comments into multiple lines.

File: src/js/port_handler.js (hunk 2 above)
Comment (Oats87) [reply src/js/port_handler.js:71]: oops

**Review by breavyn — APPROVED** [review]

**Review by rotorflight — APPROVED** [review]

**Pull request merged**